src/service/routing/directions.clj

```diff
@@ -65,21 +70,25 @@
    Example:
    (direction graph :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]"
   [graph & parameters]
-  (let [{:keys [coordinates steps]} parameters
-        start     (brute-nearest graph (first coordinates))
-        dst       (brute-nearest graph (last coordinates))
-        traversal (alg/dijkstra graph :value-by length
-                                      :start-from #{(key start)})
-        trace     (reduce (fn [_ trace] (when (= (key trace) (key dst)) (reduced trace)))
-                          nil
-                          traversal)]
-    (if (nil? trace)
-      {:code "NoRoute"}
-      {:code "Ok"
-       :waypoints (map (fn [point] {:name "wonderland"
-                                    :location [(:lon point) (:lat point)]})
-                       coordinates)
-       :routes [(route graph trace)]})))
+  (if-not (validate-radiuses parameters)
```

Comment:
Yes you are right. I guess we would have to either separate the `specs` from the API or define the radiouses completely at runtime.

I would prefer if we separate the two namespaces to avoid cyclic dependencies in the future as well.